src/statsmodels/formula.jl

```diff
@@ -485,7 +485,16 @@ creating the model matrix.
         error("Could not construct model matrix. Resulting matrix has 0 columns.")
     end
 
-    ModelMatrix{T}(reduce(hcat, blocks), assign)
+    I = size(dfrm, 1)
+    J = mapreduce(x -> size(x, 2), +, blocks)
+    X = Array(eltype(T), I, J)
```

Comment:
> blocks is a Vector{T} so blocks[1] will always be a T, i.e. correctly sparse or dense depending on what was requested.

OK. I thought that part of the code hadn't been adapted to use sparse matrices, even when requested.

> At least in my use cases, the reason for wanting a ModelMatrix{SparseMatrixCSC{Float64,Int}} is because a ModelMatrix{Matrix{Float64}} wouldn't fit in memory... Thus an Array{eltype(T)}(I, J) wouldn't fit in memory either, defeating the point of supporting sparse ModelMatrix construction in the first place.

Sure, that's really essential.